aten/src/THC/THCNumerics.cuh

```diff
@@ -309,6 +309,77 @@ struct THCNumerics<float> {
+  static inline __host__ __device__  at::BFloat16 floor(at::BFloat16 a) { return floorf(a); }
+  static inline __host__ __device__  at::BFloat16 trunc(at::BFloat16 a) { return truncf(a); }
+  static inline __host__ __device__  at::BFloat16 acos (at::BFloat16 a) { return  acosf(a); }
+  static inline __host__ __device__  at::BFloat16 cosh (at::BFloat16 a) { return  coshf(a); }
+  static inline __host__ __device__  at::BFloat16 acosh(at::BFloat16 a) { return acoshf(a); }
+  static inline __host__ __device__  at::BFloat16 asin (at::BFloat16 a) { return  asinf(a); }
+  static inline __host__ __device__  at::BFloat16 sinh (at::BFloat16 a) { return  sinhf(a); }
+  static inline __host__ __device__  at::BFloat16 asinh(at::BFloat16 a) { return asinhf(a); }
+  static inline __host__ __device__  at::BFloat16 tan  (at::BFloat16 a) { return   tanf(a); }
+  static inline __host__ __device__  at::BFloat16 atan (at::BFloat16 a) { return  atanf(a); }
+  static inline __host__ __device__  at::BFloat16 tanh (at::BFloat16 a) { return  tanhf(a); }
+  static inline __host__ __device__  at::BFloat16 erf  (at::BFloat16 a) { return   erff(a); }
+  static inline __host__ __device__  at::BFloat16 erfc (at::BFloat16 a) { return  erfcf(a); }
+  static inline __host__ __device__  at::BFloat16 abs  (at::BFloat16 a) { return  fabsf(a); }
+  static inline __host__ __device__  at::BFloat16 round(at::BFloat16 a) { return nearbyintf(a); }
+  static inline __host__ __device__  at::BFloat16 frac (at::BFloat16 a) { return a - truncf(a); }
+  static inline __host__ __device__  at::BFloat16 cinv (at::BFloat16 a) { return 1.0f / a; }
+  static inline __host__ __device__  at::BFloat16 add  (at::BFloat16 a, at::BFloat16 b) { return a + b; }
+  static inline __host__ __device__  at::BFloat16 div  (at::BFloat16 a, at::BFloat16 b) { return a / b; }
+  static inline __host__ __device__  at::BFloat16 mul  (at::BFloat16 a, at::BFloat16 b) { return a * b; }
+  static inline __host__ __device__  at::BFloat16 sub  (at::BFloat16 a, at::BFloat16 b) { return a - b; }
+  static inline __host__ __device__  at::BFloat16 pow  (at::BFloat16 a, at::BFloat16 b) { return powf(a, b); }
+  static inline __host__ __device__  at::BFloat16 atan2(at::BFloat16 a, at::BFloat16 b) { return atan2f(a, b); }
+
+  static inline __host__ __device__ bool isnan(at::BFloat16 a) {
+    #ifdef _MSC_VER
+      // Windows requires this explicit conversion. The reason is unclear
+      // related issue with clang: https://reviews.llvm.org/D37906
+      return ::isnan((float) a);
+    #else
+      return ::isnan(a);
+    #endif
+  }
+
+  static inline __host__ __device__ bool isinf(at::BFloat16 a) {
+    #ifdef _MSC_VER
+      // Windows requires this explicit conversion. The reason is unclear
+      // related issue with clang: https://reviews.llvm.org/D37906
+      return ::isinf((float) a);
+    #else
+      return ::isinf(a);
+    #endif
+  }
+};
+
```

Comment:
@ezyang, this part

Reply:
`This diff looks very familiar to me. Is it a respin of a diff you did previously?`
The diff is identical to the one that have been reverted except this part which fixes win build.